tests/controllers/cleanup/delete_tokens_test.go

```diff
@@ -0,0 +1,165 @@
+	"testing"
+	"time"
+)
+
+type AuthConfigSuite struct {
+	suite.Suite
+	ctx               context.Context
+	cancel            context.CancelFunc
+	testEnv           *envtest.Environment
+	managementContext *config.ManagementContext
+	wranglerContext   *wrangler.Context
+}
+
+func TestAuthConfigSuite(t *testing.T) {
+	suite.Run(t, new(AuthConfigSuite))
+}
+
+func (s *AuthConfigSuite) SetupSuite() {
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+
+	// Load CRD from YAML for REST Client
+	s.testEnv = &envtest.Environment{}
+	restCfg, err := s.testEnv.Start()
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), restCfg)
+
+	// Create CRDs
+	factory, err := crd.NewFactoryFromClient(restCfg)
+	assert.NoError(s.T(), err)
+
+	err = factory.BatchCreateCRDs(s.ctx,
+		crd.CRD{
+			SchemaObject: v3.Token{},
+			NonNamespace: true,
+		},
+		crd.CRD{
+			SchemaObject: v3.AuthConfig{},
+			NonNamespace: true,
+		},
+	).BatchWait()
+	assert.NoError(s.T(), err)
+
+	// Create the wrangler and management contexts
+	wranglerContext, err := wrangler.NewContext(s.ctx, nil, restCfg)
+	assert.NoError(s.T(), err)
+
+	scaledContext, clusterManager, _, err := multiclustermanager.BuildScaledContext(s.ctx, wranglerContext, &multiclustermanager.Options{})
+	assert.NoError(s.T(), err)
+	s.managementContext, err = scaledContext.NewManagementContext()
+	assert.NoError(s.T(), err)
+
+	// Register the auth controller
+	auth.RegisterEarly(s.ctx, s.managementContext, clusterManager)
+
+	// Start controllers
+	common.StartNormanControllers(s.ctx, s.T(), s.managementContext,
+		schema.GroupVersionKind{
+			Group:   "management.cattle.io",
+			Version: "v3",
+			Kind:    "AuthConfig",
+		},
+	)
+
+	// Start caches
+	common.StartWranglerCaches(s.ctx, s.T(), s.managementContext.Wrangler,
+		schema.GroupVersionKind{
+			Group:   "management.cattle.io",
+			Version: "v3",
+			Kind:    "Token",
+		},
+	)
+}
+
+func (s *AuthConfigSuite) TestTokensCleanup() {
+	t := assert.New(s.T())
+
+	const (
+		tick         = 1 * time.Second
+		duration     = 10 * time.Second
+		numOfTokens  = 1000
+		authProvider = "openldap"
+	)
+
+	// create the AuthConfig for the authProvider
+	_, err := s.managementContext.Management.AuthConfigs("").Create(&v3.AuthConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: authProvider,
+		},
+		Type:    client.OpenLdapConfigType,
+		Enabled: true,
+	})
+	t.NoError(err)
+
+	// create some tokens for the authProvider
+	for i := 0; i < numOfTokens; i++ {
+		_, err := s.managementContext.Management.Tokens("").Create(&v3.Token{
+			ObjectMeta:   metav1.ObjectMeta{GenerateName: "t-"},
+			AuthProvider: authProvider,
+		})
+		t.NoError(err)
+	}
+
+	// and also some local tokens
+	for i := 0; i < numOfTokens; i++ {
+		_, err := s.managementContext.Management.Tokens("").Create(&v3.Token{
+			ObjectMeta:   metav1.ObjectMeta{GenerateName: "t-"},
+			AuthProvider: "local",
+		})
+		t.NoError(err)
+	}
+
+	fmt.Printf("created %d tokens\n", numOfTokens)
```

Comment:
IMHO `fmt.Print` should not be used directly even in tests. If you need to output something like this t.Log can be used instead.
```suggestion
	fmt.Printf("created %d tokens\n", numOfTokens)
```

Reply:
Done with 3801ebd